### ISSUE TITLE
Cannot install luajwtjitsi 3.0-0 on 22.04

### ISSUE THREAD

**Issue by damencho** [jitsi/jitsi-meet#11490]
On clean install:
```
apt install lua-basexx
apt install luarocks
apt install liblua5.2-dev
sudo luarocks install luajwtjitsi 3.0-0
```
Get errors:
```
Installing https://luarocks.org/luaossl-20200709-0.src.rock

luaossl 20200709-0 depends on lua (5.2-1 provided by VM)
Applying patch config.h.diff...
Hunk 1 found at offset 2...
gcc -O2 -fPIC -I/usr/include/lua5.2 -c src/openssl.c -o src/openssl.o -D_REENTRANT -D_THREAD_SAFE -DCOMPAT53_PREFIX=luaossl -D_GNU_SOURCE -I/usr/include -I/usr/include
...
...
...
src/openssl.c: In function ‘luaopen__openssl’:
src/openssl.c:3266:27: error: ‘SHLIB_VERSION_HISTORY’ undeclared (first use in this function); did you mean ‘SSL3_VERSION_MINOR’?
 3266 |         lua_pushstring(L, SHLIB_VERSION_HISTORY);
      |                           ^~~~~~~~~~~~~~~~~~~~~
      |                           SSL3_VERSION_MINOR
src/openssl.c:3266:27: note: each undeclared identifier is reported only once for each function it appears in
src/openssl.c:3269:27: error: ‘SHLIB_VERSION_NUMBER’ undeclared (first use in this function); did you mean ‘SSLEAY_VERSION_NUMBER’?
 3269 |         lua_pushstring(L, SHLIB_VERSION_NUMBER);
      |                           ^~~~~~~~~~~~~~~~~~~~
      |                           SSLEAY_VERSION_NUMBER
```

**Comment by saghul** [comment]
What about lua 5.4?

**Comment by damencho** [comment]
The problem is in luaossl ...

**Comment by saghul** [comment]
Ah. Did you have libssl1.1-dev installed? I think it might not be compatible with 3.0.

**Comment by damencho** [comment]
that is not available on 22.04 :)

https://packages.ubuntu.com/search?keywords=libssl1.1-dev&searchon=names&suite=jammy&section=all

So we either need to fix luaossl or move away from it, if possible.

https://github.com/wahern/luaossl/issues/195

**Comment by saghul** [comment]
Ah sh*t.

**Comment by damencho** [comment]
Glorious one :) 💩

**Comment by damencho** [comment]
@saghul I went to ask the prosody guys for advice :) So they have https://hg.prosody.im/0.12/file/tip/util/jwt.lua which is HS256 only :(

**Comment by jprjr** [comment]
Ubuntu ships an older version of luaossl on 22.04. Not sure what the minimum requirement for luajwtjitsi is. https://packages.ubuntu.com/jammy/lua-luaossl

**Comment by damencho** [comment]
No idea why is that, but the rockspec says: "luaossl >= 20190731-0",

**Comment by jprjr** [comment]
Also want to point out I have a PR open that attempts to fix the issue: https://github.com/wahern/luaossl/pull/199

Would love to get eyes on it, get it tested, etc.

**Comment by damencho** [comment]
Thanks for letting us know :)

**Comment by damencho** [comment]
Yeah another option for us is using [lua-luaossl](https://packages.ubuntu.com/jammy/lua-luaossl) debian package with luajwtjitsi (if it works) by just creating a debian package for luajwtjitsi and drop the luarocks installation from our packages scripts.

**Comment by saghul** [comment]
> Yeah another option for us is using [lua-luaossl](https://packages.ubuntu.com/jammy/lua-luaossl) debian package with luajwtjitsi (if it works) by just creating a debian package for luajwtjitsi and drop the luarocks installation from our packages scripts. 
> 
> 

Not a fan of keeping that package but not sure there is much else we can do.

**Comment by nitin-gyany** [comment]
How to install neovim on 22.04?

**Comment by saghul** [comment]
@nitin-gyany Wrong repo, this has nothing to do with NeoVim.

**Comment by nitin-gyany** [comment]
ok. Let me try again

**Comment by nitin-gyany** [comment]
I got neovim installed on 22.04. Thank you.
